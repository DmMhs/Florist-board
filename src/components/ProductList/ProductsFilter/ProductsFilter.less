.filter-wrapper {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  width: 20%;
  border-right: 0.5px solid #eee;
  transition: 0.5s;
  z-index: 3;
  .toggle {
    color: #4d4c4c;
    font-size: 2rem;
    position: absolute;
    top: calc(50% - 1rem);
    right: -2rem;
    cursor: pointer;
    &:hover {
      color: #333333;
    }
  }
  &.hide {
    position: relative;
    transform: translateX(-100%);
    transition: 0.5s;
  }
  h2 {
    font-weight: lighter;
    background-color: #eee;
    padding: 1rem;
    color: #333333;
    display: flex;
    justify-content: space-between;
    align-items: center;
    span {
      margin-left: 1rem;
    }
    i {
      font-size: 1rem;
      &.close {
        margin-right: 1rem;
        cursor: pointer;
      }
    }
  }
  .filter-option {
    display: flex;
    flex-direction: row;
    border-bottom: 0.5px solid #eee;
    padding: 2rem;
    flex-wrap: nowrap;
    justify-content: flex-start;
    align-items: center;
    label {
      display: inline-block;
      text-align: left;
      font-weight: lighter;
      font-size: 1.2rem;
    }
    &.price-range {
      display: flex;
      flex-direction: column;
      label {
        width: 100%;
        margin-bottom: 1rem;
      }
      div {
        display: flex;
        flex-direction: row;
        justify-content: space-around;
        align-items: center;
        flex-wrap: nowrap;
        font-size: 1.5rem;
        color: #333333;
        input {
          width: 43%;
          font-size: 1.5rem;
          color: #333333;
          font-weight: lighter;
          text-align: center;
          border-radius: 5px;
          border: 1px solid #eee;
        }
      }
    }
    input[type='checkbox'] {
      width: 50%;
    }
  }
}
@media screen and (max-width: 920px) {
  .filter-wrapper {
    position: fixed;
    display: flex;
    width: 100%;
    top: 2.5rem;
    bottom: 0;
    transition: 0;
    border-right: none;
    background-color: white;
    &.hide {
      transform: translateX(0);
      width: 0;
      transform: none;
      transition: 0;
      border: none;
      display: none;
    }
    .filter-option {
      &.price-range {
        div {
          font-size: 1.2rem;
          input {
            font-size: 1.2rem;
          }
        }
      }
    }
  }
}
@media screen and (min-width: 768.1px) and (max-width: 1200px) {
  .filter-wrapper {
    width: 25%;
  }
}
