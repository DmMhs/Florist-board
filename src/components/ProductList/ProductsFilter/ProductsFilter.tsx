import React from 'react';

import './ProductsFilter.less';
import labels from '../../../config/labels';

interface ProductsFilterProps {
  filterToggle:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
  filtersSidebarRef: React.RefObject<HTMLDivElement>;
  filterToggleRef: React.RefObject<HTMLDivElement>;
  inStockChanged:
    | ((event: React.ChangeEvent<HTMLInputElement>) => void)
    | undefined;
  priceFromChanged:
    | ((event: React.ChangeEvent<HTMLInputElement>) => void)
    | undefined;
  priceToChanged:
    | ((event: React.ChangeEvent<HTMLInputElement>) => void)
    | undefined;
  mobileMode?: boolean;
}

const productsFilter = (props: ProductsFilterProps) => {
  const closeFiltersIcon =
    props.mobileMode === true ? (
      <i className="fas fa-times close" onClick={props.filterToggle} />
    ) : null;
  return (
    <div className="filter-wrapper hide" ref={props.filtersSidebarRef}>
      {props.mobileMode === false ? (
        <i
          className="fas fa-filter toggle"
          onClick={props.filterToggle}
          ref={props.filterToggleRef}
        />
      ) : null}
      <h2>
        <span>
          {labels.pages.shop.filter.main} <i className="fas fa-filter" />
        </span>{' '}
        {closeFiltersIcon}
      </h2>
      <form className="filter-form">
        <div className="filter-option available">
          <label>{labels.pages.shop.filter.available}</label>
          <input type="checkbox" onChange={props.inStockChanged} />
        </div>
        <div className="filter-option price-range">
          <label>{labels.pages.shop.filter.price}</label>
          <div>
            <input
              type="number"
              onChange={props.priceFromChanged}
              className="priceFrom"
              placeholder={labels.pages.shop.filter.priceInputs.from}
              min="0"
            />
            -
            <input
              type="number"
              onChange={props.priceToChanged}
              className="priceTo"
              placeholder={labels.pages.shop.filter.priceInputs.to}
              min="0"
            />
          </div>
        </div>
      </form>
    </div>
  );
};

export default productsFilter;
