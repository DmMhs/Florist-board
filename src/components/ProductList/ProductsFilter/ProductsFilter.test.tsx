import React from 'react';
import { shallow } from 'enzyme';

import ProductsFilter from './ProductsFilter';

it('matches a snapshot', () => {
  const wrapper = shallow(<ProductsFilter />);
  expect(wrapper).toMatchSnapshot();
});
