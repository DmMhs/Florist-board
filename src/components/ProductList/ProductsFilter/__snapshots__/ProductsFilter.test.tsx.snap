// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="filter-wrapper hide"
>
  <h2>
    <span>
      Filters 
      <i
        className="fas fa-filter"
      />
    </span>
     
  </h2>
  <form
    className="filter-form"
  >
    <div
      className="filter-option available"
    >
      <label>
        IN STOCK
      </label>
      <input
        type="checkbox"
      />
    </div>
    <div
      className="filter-option price-range"
    >
      <label>
        PRICE RANGE
      </label>
      <div>
        <input
          className="priceFrom"
          min="0"
          placeholder="from"
          type="number"
        />
        -
        <input
          className="priceTo"
          min="0"
          placeholder="to"
          type="number"
        />
      </div>
    </div>
  </form>
</div>
`;
